pyClient/zethUtils.py

```diff
@@ -19,61 +18,43 @@
-   Asiacrypt 2001,
-   <https://iacr.org/archive/asiacrypt2001/22480568.pdf>
-"""
-def encrypt(message, public_key):
-    rsa_key = RSA.importKey(public_key)
-    rsa_key = PKCS1_OAEP.new(rsa_key)
-
-    blob = zlib.compress(message.encode())
-
-    # Refer to: https://pycryptodome.readthedocs.io/en/latest/src/cipher/oaep.html#Crypto.Cipher.PKCS1_OAEP.PKCS1OAEP_Cipher.encrypt
-    # to define the chunk size
-    chunk_size = 62 # (128 - 2 - 2*32) since we use RSA modulus of 1024 bits (128 bytes) in the keystore
-    offset = 0
-    end_loop = False
-    encrypted =  "".encode()
-
-    while not end_loop:
-        chunk = blob[offset:offset + chunk_size]
-
-        # Padding
-        if len(chunk) % chunk_size != 0:
-            end_loop = True
-            chunk += " ".encode() * (chunk_size - len(chunk))
-
-        encrypted += rsa_key.encrypt(chunk)
-        offset += chunk_size
-
-    return base64.b64encode(encrypted)
-
-def decrypt(encrypted_blob, private_key):
-    rsakey = RSA.importKey(private_key)
-    rsakey = PKCS1_OAEP.new(rsakey)
-
-    encrypted_blob = base64.b64decode(encrypted_blob)
-
-    chunk_size = 128 # Size of the modulus we use here
-    offset = 0
-    decrypted = "".encode()
-
-    # Loop over our chunks
-    while offset < len(encrypted_blob):
-        chunk = encrypted_blob[offset: offset + chunk_size]
-        decrypted += rsakey.decrypt(chunk)
-        offset += chunk_size
-    return str(zlib.decompress(decrypted), 'utf-8')
+# Gets PrivateKey object from hexadecimal representation (see: https://pynacl.readthedocs.io/en/stable/public/#nacl.public.PrivateKey)
+def get_private_key_from_bytes(private_key_bytes):
+  return PrivateKey(private_key_bytes, encoder=nacl.encoding.RawEncoder)
+
+# Gets PublicKey object from hexadecimal representation (see: https://pynacl.readthedocs.io/en/stable/public/#nacl.public.PublicKey)
+def get_public_key_from_bytes(public_key_bytes):
+  return PublicKey(public_key_bytes, encoder=nacl.encoding.RawEncoder)
+
+# Encrypts a string message by using valid ec25519 public key (bytes) and a private key object. See: https://pynacl.readthedocs.io/en/stable/public/
+def encrypt(message, public_key_bytes, private_key):
+  # Decodes hex representation to keys objects
+  public_key = get_public_key_from_bytes(public_key_bytes)
+
+  # Inits encryption box instance
+  encryption_box = Box(private_key, public_key)
+
+  # Encods str message to bytes
+  message_bytes = message.encode('utf-8')
+
+  # Encrypts the message. The nonce is chosen randomly.
+  encrypted = encryption_box.encrypt(message_bytes, encoder=nacl.encoding.RawEncoder)
+
+  # Need to cast to the parent class Bytes of nacl.utils.EncryptedMessage to make it accepted from mix solidity function
+  return bytes(encrypted)
+
+# Decrypts a string ciphertext by using valid ec25519 public key (bytes) and a private key object. See: https://pynacl.readthedocs.io/en/stable/public/
+def decrypt(encrypted_message, public_key_bytes, private_key):
```

Comment:
As mentioned in previous comment, change to:
`def decrypt(encrypted_message, public_key, private_key):` and do the conversion from pk_bytes to `pk` before.
Another alternative would be to have:
`def decrypt(encrypted_message, public_key_bytes, private_key_bytes):` and to convert both keys from bytes to the appropriate type inside the function. Both would work, but now it's not consistent.

Reply:
Moreover, it might be useful to rename variable to inform the provenance of the key. Ie: For decryption, the `public_key` comes from the sender. so I would rename: `pubkey_sender`, and `privkey_recipient`, and vice-versa for encryption. This makes it clearer which key we're talking about.

Reply:
Variables renamed all along the cryptographic primitives: `encrypt/decrypt/receive` and tests.